mlflow/utils/autologging_utils/safety.py

```diff
@@ -24,6 +25,13 @@
 
 _AUTOLOGGING_PATCHES = {}
 
+_MUTABLE_ARGS_ALLOWLIST = {
```

Comment:
Can we add a comment above this list explaining what it's for? I.e for specifying a list of function arguments that can be mutated by autologging code. We should also warn developers that this list shouldn't be modified unless absolutely necessary and that, if modified, a clear reason needs to be provided as an inline comment + thorough integration tests need to be introduced.